src/_guides/language/effective-dart/design.md

```diff
@@ -1087,8 +1354,8 @@ void log(Object object) {
   print(object.toString());
 }
 
-// Only accepts bool or String, which can't be expressed in a type annotation.
-bool convertToBool(arg) {
+// Only accepts bool or String, which type system can't express.
+bool convertToBool(dynamic arg) {
   if (arg is bool) return arg;
   if (arg is String) return arg == 'true';
   throw new ArgumentError('Cannot convert $arg to a bool.');
```

Comment:
_Maybe_:

```dart
  bool convertToBool(dynamic arg) => arg is bool
      ? arg
      : arg is String
          ? arg == 'true'
          : throw new ArgumentError('bool or String expected, got $arg');
```

Reply:
Nested ternary operators in a style guide? I'll pass. 😄

Reply:
McCarthy conditionals FTW! ;)